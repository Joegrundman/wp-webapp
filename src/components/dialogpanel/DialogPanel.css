.panel {
  position: fixed;
  width: 20rem;
  height: 100%;
  top: 60px;
  left: 60px;
  background-color: #2a2a2a;
  z-index: 3;
  overflow-x: hidden;
}