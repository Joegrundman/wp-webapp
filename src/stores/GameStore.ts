import { URL_MAP_EUR, URL_MAP_PAC } from 'Constants/ui-constants'
import { observable } from 'mobx'

class GameStore {

  @observable public theater: string = URL_MAP_EUR
  @observable public showDialogPanel: boolean = false
  @observable public currentActiveDialog: string = ''

  public setTheater (theater: string): void {
    this.theater = theater
  }

  public toggleTheater (): void {
    if (this.theater === URL_MAP_EUR) {
      this.theater = URL_MAP_PAC
    } else {
      this.theater = URL_MAP_EUR
    }
  }

  public toggleDialogPanel (): void {
    this.showDialogPanel = !this.showDialogPanel
  }

  public openDialogPanel (): void {
    this.showDialogPanel = true
  }

  public handleDialogButton(dialog: string): void {
    if(this.showDialogPanel && this.currentActiveDialog !== dialog) {
      this.setCurrentActiveDialog(dialog)
    } else if(this.showDialogPanel && this.currentActiveDialog === dialog) {
      this.closeDialogPanel()
      this.setCurrentActiveDialog('')
    } else {
      this.openDialogPanel()
      this.setCurrentActiveDialog(dialog)
    }  
  }

  public closeDialogPanel (): void {
    this.showDialogPanel = false
  }

  public setCurrentActiveDialog (dialog: string): void {
    this.currentActiveDialog = dialog
  }
}

export default GameStore