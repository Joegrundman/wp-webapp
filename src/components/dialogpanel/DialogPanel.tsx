import { observer } from 'mobx-react'
import * as React from 'react'
import GameStore from 'Stores/GameStore'
import Shipyard from '../shipyard/Shipyard'
import Taskforce from '../taskforce/Taskforce'
import locals from './DialogPanel.css'

export interface IDialogPanelProps {
  store: GameStore
}

@observer
class DialogPanel extends React.Component<IDialogPanelProps> {

  public getDialog(): JSX.Element | '' {
    switch(this.props.store.currentActiveDialog) {
      case 'shipyard': return (<Shipyard />)
      case 'taskforce': return (<Taskforce />)
      default: return ''
    }
  }

  public render (): JSX.Element {
    const { showDialogPanel } = this.props.store
    return (
      <div className={locals.panel} style={{display: showDialogPanel ? '': 'none' }}>
        {this.getDialog()}
      </div> 
    )
  }
}

export default DialogPanel
