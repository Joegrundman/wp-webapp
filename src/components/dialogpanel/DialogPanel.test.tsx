import { shallow, ShallowWrapper } from 'enzyme'
import * as React from 'react'
import GameStore from 'Stores/GameStore'
import DialogPanel, { IDialogPanelProps } from './DialogPanel'

describe('<DialogPanel />', () => {

 let mockGame: GameStore
 let defaultProps: IDialogPanelProps

  beforeEach(() => {
    mockGame = new GameStore()
    mockGame.openDialogPanel()
    defaultProps = {
      store: mockGame
     }
  })

  it('should render as expected with defaultProps', () => {
    const component: ShallowWrapper = shallow(<DialogPanel {...defaultProps}/>)
    expect(component).toMatchSnapshot()
  })

  it('should render as expected with currentActiveDialog set to shipyard', () => {
    mockGame.setCurrentActiveDialog('shipyard')
    const component: ShallowWrapper = shallow(<DialogPanel {...defaultProps}/>)
    expect(component).toMatchSnapshot()
  })

  it('should render as expected with showDialogPanel set to false', () => {
    mockGame.closeDialogPanel()
    const component: ShallowWrapper = shallow(<DialogPanel {...defaultProps}/>)
    expect(component).toMatchSnapshot()
  })

  it('should render as expected with currentActiveDialog set to taskforce', () => {
    mockGame.setCurrentActiveDialog('taskforce')
    const component: ShallowWrapper = shallow(<DialogPanel {...defaultProps}/>)
    expect(component).toMatchSnapshot()
  })

})
