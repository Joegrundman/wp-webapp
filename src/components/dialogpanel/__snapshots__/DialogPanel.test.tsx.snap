// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DialogPanel /> should render as expected with currentActiveDialog set to shipyard 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DialogPanel
    store={
      GameStore {
        "currentActiveDialog": "shipyard",
        "showDialogPanel": true,
        "theater": "http://res.cloudinary.com/druzhkwmt/image/upload/v1456179368/b3oj0apypmsiu1zsypwl.jpg",
        Symbol(mobx did run lazy initializers): true,
        Symbol(mobx administration): ObservableObjectAdministration$$1 {
          "defaultEnhancer": [Function],
          "keysAtom": Atom$$1 {
            "diffValue": 0,
            "isBeingObserved": false,
            "isPendingUnobservation": false,
            "lastAccessedBy": 0,
            "lowestObserverState": 2,
            "name": "GameStore@3.keys",
            "observers": Set {},
          },
          "name": "GameStore@3",
          "target": [Circular],
          "values": Map {
            "theater" => "http://res.cloudinary.com/druzhkwmt/image/upload/v1456179368/b3oj0apypmsiu1zsypwl.jpg",
            "showDialogPanel" => true,
            "currentActiveDialog" => "shipyard",
          },
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Shipyard />,
      "className": undefined,
      "style": Object {
        "display": "",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Shipyard />,
        "className": undefined,
        "style": Object {
          "display": "",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<DialogPanel /> should render as expected with currentActiveDialog set to taskforce 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DialogPanel
    store={
      GameStore {
        "currentActiveDialog": "taskforce",
        "showDialogPanel": true,
        "theater": "http://res.cloudinary.com/druzhkwmt/image/upload/v1456179368/b3oj0apypmsiu1zsypwl.jpg",
        Symbol(mobx did run lazy initializers): true,
        Symbol(mobx administration): ObservableObjectAdministration$$1 {
          "defaultEnhancer": [Function],
          "keysAtom": Atom$$1 {
            "diffValue": 0,
            "isBeingObserved": false,
            "isPendingUnobservation": false,
            "lastAccessedBy": 0,
            "lowestObserverState": 2,
            "name": "GameStore@7.keys",
            "observers": Set {},
          },
          "name": "GameStore@7",
          "target": [Circular],
          "values": Map {
            "theater" => "http://res.cloudinary.com/druzhkwmt/image/upload/v1456179368/b3oj0apypmsiu1zsypwl.jpg",
            "showDialogPanel" => true,
            "currentActiveDialog" => "taskforce",
          },
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Taskforce />,
      "className": undefined,
      "style": Object {
        "display": "",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Taskforce />,
        "className": undefined,
        "style": Object {
          "display": "",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<DialogPanel /> should render as expected with defaultProps 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DialogPanel
    store={
      GameStore {
        "currentActiveDialog": "",
        "showDialogPanel": true,
        "theater": "http://res.cloudinary.com/druzhkwmt/image/upload/v1456179368/b3oj0apypmsiu1zsypwl.jpg",
        Symbol(mobx did run lazy initializers): true,
        Symbol(mobx administration): ObservableObjectAdministration$$1 {
          "defaultEnhancer": [Function],
          "keysAtom": Atom$$1 {
            "diffValue": 0,
            "isBeingObserved": false,
            "isPendingUnobservation": false,
            "lastAccessedBy": 0,
            "lowestObserverState": 2,
            "name": "GameStore@1.keys",
            "observers": Set {},
          },
          "name": "GameStore@1",
          "target": [Circular],
          "values": Map {
            "theater" => "http://res.cloudinary.com/druzhkwmt/image/upload/v1456179368/b3oj0apypmsiu1zsypwl.jpg",
            "showDialogPanel" => true,
            "currentActiveDialog" => "",
          },
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "",
      "className": undefined,
      "style": Object {
        "display": "",
      },
    },
    "ref": null,
    "rendered": "",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "",
        "className": undefined,
        "style": Object {
          "display": "",
        },
      },
      "ref": null,
      "rendered": "",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<DialogPanel /> should render as expected with showDialogPanel set to false 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DialogPanel
    store={
      GameStore {
        "currentActiveDialog": "",
        "showDialogPanel": false,
        "theater": "http://res.cloudinary.com/druzhkwmt/image/upload/v1456179368/b3oj0apypmsiu1zsypwl.jpg",
        Symbol(mobx did run lazy initializers): true,
        Symbol(mobx administration): ObservableObjectAdministration$$1 {
          "defaultEnhancer": [Function],
          "keysAtom": Atom$$1 {
            "diffValue": 0,
            "isBeingObserved": false,
            "isPendingUnobservation": false,
            "lastAccessedBy": 0,
            "lowestObserverState": 2,
            "name": "GameStore@5.keys",
            "observers": Set {},
          },
          "name": "GameStore@5",
          "target": [Circular],
          "values": Map {
            "theater" => "http://res.cloudinary.com/druzhkwmt/image/upload/v1456179368/b3oj0apypmsiu1zsypwl.jpg",
            "showDialogPanel" => false,
            "currentActiveDialog" => "",
          },
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "",
      "className": undefined,
      "style": Object {
        "display": "none",
      },
    },
    "ref": null,
    "rendered": "",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "",
        "className": undefined,
        "style": Object {
          "display": "none",
        },
      },
      "ref": null,
      "rendered": "",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
